### PR TITLE
Delete auto-generated annotations when job is deleted.

### LINKED ISSUE #57976: [ML] Delete job should delete associated annotations
Currently annotations are not deleted if a user deletes an ML job.

The historical decisions that led to this decision are in #33376.  For example:

> This makes the deletion logic quite complicated - annotations should be deleted when all jobs they relate to are deleted, but the wide variety of ways of specifying which jobs an annotation relates to makes determination of this complex.

and:

> We also decided that it would be useful to have persistent annotations, that _don't_ get deleted if all the jobs they relate to get deleted. This would be useful for the case of repeatedly deleting and recreating a job to try out slightly different configurations.
>
> So there should be an `is_persistent` boolean field in the annotations too.

The way annotations was eventually developed means the first thought is completely irrelevant.  We envisaged annotations that would apply to multiple jobs but currently every annotation is bound to exactly one job.

The second idea is still potentially relevant, but we never implemented the `is_persistent` boolean field.

Given that large numbers of annotations will now be generated by model change events, it makes sense that we delete all annotations related to a job with a `create_username` of `_xpack` when the job is deleted.

This should be done as an extra step in the chain of deletion steps in `TransportDeleteJobAction.deleteJobDocuments()`.

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR makes the auto-generated annotations associated with the job deleted when the job itself gets deleted.

Closes https://github.com/elastic/elasticsearch/issues/57976

### CODE REVIEW

**Review by benwtrent — APPROVED** [review]

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java [hunk 1]
```diff
@@ -444,25 +451,18 @@ private void deleteJobDocuments(ParentTaskAssigningClient parentTaskClient, Stri
...
     private void deleteQuantiles(ParentTaskAssigningClient parentTaskClient, String jobId, ActionListener<Boolean> finishedHandler) {
-        // The quantiles type and doc ID changed in v5.5 so delete both the old and new format
-        DeleteByQueryRequest request = new DeleteByQueryRequest(AnomalyDetectorsIndex.jobStateIndexPattern());
         // Just use ID here, not type, as trying to delete different types spams the logs with an exception stack trace
         IdsQueryBuilder query = new IdsQueryBuilder().addIds(Quantiles.documentId(jobId));
-        request.setQuery(query);
-        request.setIndicesOptions(MlIndicesUtils.addIgnoreUnavailable(IndicesOptions.lenientExpandOpen()));
-        request.setAbortOnVersionConflict(false);
-        request.setRefresh(true);
+        // The quantiles type and doc ID changed in v5.5 so delete both the old and new format
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java:456]: I know you've just copied it from above, but this comment is no longer correct - we are only deleting the most recent ID format.  So please delete this comment.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java [hunk 2]
```diff
@@ -479,14 +479,14 @@ private void deleteModelState(ParentTaskAssigningClient parentTaskClient, String
...
                                         ActionListener<Boolean> finishedHandler) {
-        // The categorizer state type and doc ID changed in v5.5 so delete both the old and new format
-        DeleteByQueryRequest request = new DeleteByQueryRequest(AnomalyDetectorsIndex.jobStateIndexPattern());
         // Just use ID here, not type, as trying to delete different types spams the logs with an exception stack trace
         IdsQueryBuilder query = new IdsQueryBuilder().addIds(CategorizerState.documentId(jobId, docNum));
-        request.setQuery(query);
-        request.setIndicesOptions(MlIndicesUtils.addIgnoreUnavailable(IndicesOptions.lenientExpandOpen()));
-        request.setAbortOnVersionConflict(false);
-        request.setRefresh(true);
+        // The categorizer state type and doc ID changed in v5.5 so delete both the old and new format
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java:484]: As above, it's no longer true that we're deleting the 5.4 ID format, so please delete this comment.

**Review by droberts195 — APPROVED** [review]
LGTM

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java]: Done.

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java (hunk 2 above)
Comment (przemekwitek) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteJobAction.java]: Done.

**Pull request merged**